src/test/java/org/apache/sling/bundleresource/impl/BundleResourceTest.java

```diff
@@ -111,7 +111,7 @@ void addContent(BundleResourceCache cache, String path, String content) throws I
         assertEquals(JcrConstants.NT_FILE, rsrc.getResourceType());
         assertNull(rsrc.getResourceSuperType());
         final ValueMap vm = rsrc.getValueMap();
-        assertEquals(JcrConstants.NT_FILE, vm.get(ResourceResolver.PROPERTY_RESOURCE_TYPE, String.class));
+        assertNull(vm.get(ResourceResolver.PROPERTY_RESOURCE_TYPE, String.class));
         assertEquals("foo", vm.get("test", String.class));
```

Comment:
Another test case for JSON resources would be nice, which asserts that explicit "sling:property" keys from the JSON also appear in the value map. @michalcukierman Can you extend an existing test or add a new one to cover that?

Reply:
Tests added